modules/client-handler/src/integrationTest/java/org/apache/ignite/client/handler/ItClientHandlerTest.java

```diff
@@ -167,7 +167,7 @@ void testHandshakeInvalidVersionReturnsError() throws Exception {
             final var err = unpacker.unpackString();
             
             assertArrayEquals(MAGIC, magic);
-            assertEquals(31, len);
+            assertEquals(32, len);
```

Comment:
Why this change occurred?

Reply:
Please see comments in `ClientMessagePacker#packString`. Basically we reserve string header size pessimistically to avoid extra string scan. It is better to write 1 extra byte than to spend time on precise utf8 byte length calculation.

The code from the library that we've used before was doing `byte[] bytes = str.getBytes()` and then copied those bytes to the target buffer, which involves an extra allocation and an extra copy.

Reply:
Okay then, I just wanted to make sure it won't break compatibility with other implementation

Reply:
Yes, this is compatible. We can use bigger varint format even if the number if small (e.g. number 5 can be encoded as 1 byte, but it is allowed to use 2-byte or 4-byte formats too).